### PR TITLE
Schedule loading L0 segments first

### PR DESCRIPTION
**Pull request by yah01**
/kind feature
related #27349

### CODE REVIEW

File: internal/querycoordv2/checkers/segment_checker.go [hunk 1]
```diff
@@ -207,17 +208,26 @@ func (c *SegmentChecker) getHistoricalSegmentDiff(
 		return
 	}
 	dist := c.getHistoricalSegmentsDist(replica)
-	distMap := typeutil.NewUniqueSet()
+	sort.Slice(dist, func(i, j int) bool {
+		return dist[i].Version < dist[j].Version
+	})
+	distMap := typeutil.NewConcurrentMap[int64, int64]()
```
Comment (congqixia) [inline internal/querycoordv2/checkers/segment_checker.go:214]: why use concurrent map here? this map is used only in this method

File: internal/querycoordv2/checkers/segment_checker.go [hunk 2]
```diff
@@ -207,17 +208,26 @@ func (c *SegmentChecker) getHistoricalSegmentDiff(
...
+		distMap.Insert(s.GetID(), s.Node)
 	}
 
 	nextTargetMap := c.targetMgr.GetHistoricalSegmentsByCollection(collectionID, meta.NextTarget)
 	currentTargetMap := c.targetMgr.GetHistoricalSegmentsByCollection(collectionID, meta.CurrentTarget)
 
 	// Segment which exist on next target, but not on dist
 	for segmentID, segment := range nextTargetMap {
-		if !distMap.Contain(segmentID) {
+		leader := c.dist.LeaderViewManager.GetLatestLeadersByShard(segment.GetInsertChannel())
```
Comment (congqixia) [inline internal/querycoordv2/checkers/segment_checker.go:224]: this leader could be a leader from other replica

**Review by congqixia — CHANGES_REQUESTED** [review]

**Comment by yah01** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**